File: python/leetcode/056.py
class Solution:
    def merge(self, intervals):
        if not intervals:
            return []
        res = []
        for item in sorted(intervals, key=lambda x: x.start):
            if res and item.start <= res[-1].end:
                res[-1].end = max(res[-1].end, item.end)
            else:
                res.append(item)
        return res

File: cpp/leetcode/056.cpp
class Solution {
public:
    vector<Interval> merge(vector<Interval>& intervals) {
        sort(intervals.begin(), intervals.end(), [](Interval a, Interval b) {return a.start < b.start;});
        vector<Interval> res;
        for (Interval item : intervals) {
            if (!res.empty() && res.back().end >= item.start) {
                res.back().end = max(res.back().end, item.end);
            } else {
                res.push_back(item);
            }
        }
        return res;
    }
};